build.rs

```diff
@@ -0,0 +1,37 @@
+extern crate pkg_config;
+
+use std::fs;
+use std::path::Path;
+use std::env;
+
+fn main() {
+    configure_snappy();
+}
+
+fn configure_snappy() {
+    // try pkg_config first
+    if pkg_config::find_library("snappy").is_ok() {
+        return;
+    }
+    if env::var_os("SNAPPY_STATIC").is_some() {
+        println!("cargo:rustc-link-lib=static=snappy");
+        println!("cargo:rustc-flags=-l c++");
+    } else {
+        println!("cargo:rustc-link-lib=dylib=snappy");
+    };
+
+    for f in vec!["/usr/lib","/usr/local/lib"] {
+        if is_file_in("libsnappy.a", Path::new(f)) {
+            println!("cargo:rustc-link-search={}", f);
+        }
+    }
```

Comment:
What about the following instead of hard-coding the directories:

```
    if let Some(path) = env::var_os("LD_LIBRARY_PATH") {
        for p in env::split_paths(&path) {
            if is_file_in("libsnappy.a", p.as_path()) {
                println!("cargo:rustc-link-search={}", p.display());
            }
        }
    }
```

?